controllers/ocsinitialization/ocsinitialization_controller.go

```diff
@@ -252,14 +256,75 @@ func (r *OCSInitializationReconciler) ensureRookCephOperatorConfigExists(initial
 			Namespace: initialData.Namespace,
 		},
 	}
-	err := r.Client.Create(r.ctx, rookCephOperatorConfig)
-	if err != nil && !errors.IsAlreadyExists(err) {
-		r.Log.Error(err, fmt.Sprintf("Failed to create %s configmap", util.RookCephOperatorConfigName))
+
+	opResult, err := ctrl.CreateOrUpdate(r.ctx, r.Client, rookCephOperatorConfig, func() error {
+
+		csiPluginDefaults := defaults.DaemonPlacements[defaults.CsiPluginKey]
+		csiProvisionerDefaults := defaults.DaemonPlacements[defaults.CsiProvisionerKey]
+
+		csiPluginTolerations := r.getCsiTolerations(defaults.CsiPluginKey)
+		csiProvisionerTolerations := r.getCsiTolerations(defaults.CsiProvisionerKey)
+
+		if rookCephOperatorConfig.Data == nil {
+			rookCephOperatorConfig.Data = make(map[string]string)
+		}
+
+		if err := updateTolerationsConfigFunc(rookCephOperatorConfig,
+			csiPluginTolerations, csiPluginDefaults.Tolerations, "CSI_PLUGIN_TOLERATIONS",
+			&initialData.Status.RookCephOperatorConfig.CsiPluginTolerationsModified); err != nil {
+			return err
+		}
+
+		err := updateTolerationsConfigFunc(rookCephOperatorConfig,
+			csiProvisionerTolerations, csiProvisionerDefaults.Tolerations, "CSI_PROVISIONER_TOLERATIONS",
+			&initialData.Status.RookCephOperatorConfig.CsiProvisionerTolerationsModified)
```

Comment:
For readability and locality sake, can we group code relating to the same concern and treat both calls symmetrically? 
```suggestion
	 	if rookCephOperatorConfig.Data == nil {
			rookCephOperatorConfig.Data = make(map[string]string)
		}

		csiPluginDefaults := defaults.DaemonPlacements[defaults.CsiPluginKey]
		csiPluginTolerations := r.getCsiTolerations(defaults.CsiPluginKey)
		if err := updateTolerationsConfigFunc(rookCephOperatorConfig,
			csiPluginTolerations, csiPluginDefaults.Tolerations, "CSI_PLUGIN_TOLERATIONS",
			&initialData.Status.RookCephOperatorConfig.CsiPluginTolerationsModified); err != nil {
			return err
		}

		csiProvisionerDefaults := defaults.DaemonPlacements[defaults.CsiProvisionerKey]
		csiProvisionerTolerations := r.getCsiTolerations(defaults.CsiProvisionerKey)
		if err := updateTolerationsConfigFunc(rookCephOperatorConfig,
			csiProvisionerTolerations, csiProvisionerDefaults.Tolerations, "CSI_PROVISIONER_TOLERATIONS",
			&initialData.Status.RookCephOperatorConfig.CsiProvisionerTolerationsModified) err != nil {
			return err
		}
		
		return nil
```

Reply:
Done